### ISSUE TITLE
Possible bug in DCANBoldProc that may or may not be relevant to XCP-D

### ISSUE THREAD

**Issue by LuciMoore** [PennLINC/xcp_d#1272]
@cindyhfls noticed a possible bug in our code base for DCANBoldProc. Our team is going to check it out soon. I wanted to share in case the issue is present in XCP-D as well! 

See: https://github.com/DCAN-Labs/dcan_bold_processing/issues/28

**Comment by cindyhfls** [comment]
At first glance, the arc length conversion and FD calculation with the brain radius specification problem in dcan-bold-processing were solved in XPC-D.

The only other thing remaining is the filter design. First of all, the iirnotch() designs a 2nd order IIR notch digital filter (https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.iirnotch.html) and filtfilt() filters forward and backward so one application already brings the order = 4 (default filter order), so applying the filter twice would bring this to order = 8. 

Additionally for notch filter is good for removing the central frequency but for the respiration band, people may have different respiration rate so it would make more sense to remove the whole band with a butterworth butter() filter?

Also only tested in MATLAB code but increasing the order manually seems to affect neighboring frequencies and not just the current frequency. chatgpt suggested the use of sosfiltfilt() function for numerical stability. Or if we use butter() function we can directly state the filter order.

I will try to submit a pull request for the actual code when I get the time.

![image](https://github.com/user-attachments/assets/3967de9e-f859-43c9-8f1c-1f1357604f74)

**Comment by cindyhfls** [comment]
![image](https://github.com/user-attachments/assets/6860e01c-91ec-438f-9d1b-f82a1376615a)

**Comment by cindyhfls** [comment]
![image](https://github.com/user-attachments/assets/7e02a686-7757-4295-947f-de5a4e55349c)

**Comment by tsalo** [comment]
Thanks @cindyhfls!

**Comment by nimzodisaster** [comment]
> good for removing the central frequency but for the respiration band, people may have different respiration rate so it would make more sense to remove the whole band with a butterworth butter() filter?
> 
> Also only tested in MATLAB code but increasing the order manually seems to affect neighboring frequencies and not just the current frequency. 

I don't know how Powers originally decided on 50mm for sure, but I note that the volume of the MNI152NLin2009cAsym brain mask is 1,827,243 mm^3, which using the formula for radius of a sphere of that volume, the radius is 75.9mm. Visually inspecting the MNI atlas, I note that 50mm that Powers chose typically spans from the center of the brain to a bit over the sulcus depth., not the surface, and this makes sense to me. 
![image](https://github.com/user-attachments/assets/a74a8b77-3cd8-4f71-bcb3-083cbde754d4)

Thus I applied a correction factor that takes the MNI to sphere radius of 75 to 50 mm, which means that I multiplied my radius estimate to 50/~75.8...or just 2/3. This came up for me because I am processing data from 2 year olds and  their radius estimates were always more than 50mm, which didn't seem right.
